libs/platform/ebpf_serialize.c

```diff
@@ -179,3 +214,384 @@ ebpf_deserialize_map_information_array(
 
     return result;
 }
+
+void
+ebpf_program_information_free(_Pre_maybenull_ _Post_invalid_ _In_ ebpf_program_information_t* program_info)
+{
+    if (program_info != NULL) {
+        ebpf_free(program_info->program_type_descriptor.context_descriptor);
+        ebpf_free((void*)program_info->program_type_descriptor.name);
```

Comment:
curious... I thought you could pass any pointer into a void* argument without complaint, why is the cast needed?

Reply:
I got a compiler warning (treated as error) due to the cosnt qualifier in program_type_descriptor.name field.

```
F:\ebpf-for-windows\my\libs\platform\ebpf_serialize.c(225,61): warning C4090: 'function': different 'const' qu
       alifiers [F:\ebpf-for-windows\my\libs\platform\user\platform_user.vcxproj]

typedef struct _ebpf_program_type_descriptor
    {
    MIDL([string])
    const char* name;
    ebpf_context_descriptor_t *context_descriptor;
    GUID platform_specific_data;
    unsigned char is_privileged;
    } 	ebpf_program_type_descriptor_t;
```